### PR TITLE
[PAPI-244] Wallet position improvements

### PR DESCRIPTION
**Pull request by drmathias**
Replaces `IncludeLpTokens` filter with `TokenType` that allows balances to be filtered by: `NonProvisional`, `Provisional` and `All`.

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,40 @@
+{
```
Comment (mrtpain) [inline .vscode/launch.json:1]: Probably fine to commit these .vscode files other option is just to modify .gitignore 

Up to you, I'm fine with either

File: src/Opdex.Platform.Infrastructure/Data/Handlers/Addresses/Balances/SelectAddressBalancesWithFilterQueryHandler.cs [hunk 2]
```diff
@@ -58,7 +58,7 @@ public async Task<IEnumerable<AddressBalance>> Handle(SelectAddressBalancesWithF
         {
             var balanceId = request.Cursor.Pointer;
 
-            var queryParams = new SqlParams(balanceId, request.Address, request.Cursor.Tokens, request.Cursor.IncludeLpTokens);
+            var queryParams = new SqlParams(balanceId, request.Address, request.Cursor.Tokens, request.Cursor.TokenType == TokenProvisionalFilter.All || request.Cursor.TokenType == TokenProvisionalFilter.NonProvisional);
```
Comment (mrtpain) [inline src/Opdex.Platform.Infrastructure/Data/Handlers/Addresses/Balances/SelectAddressBalancesWithFilterQueryHandler.cs:61]: I _think_ this should be `== TokenProvisionalFilter.NonProvisional` to set `IncludeLpTokens` to true when filtering for LP tokens. DIdn't test it but looks that way to me.

**Comment by mrtpain** [comment]
LGTM for the most part with one comment to double check first.

File: .vscode/launch.json (hunk 1 above)
Comment (drmathias) [reply .vscode/launch.json:1]: Committing them should make it easier to run locally, it's intentional.

File: src/Opdex.Platform.Infrastructure/Data/Handlers/Addresses/Balances/SelectAddressBalancesWithFilterQueryHandler.cs (hunk 2 above)
Comment (drmathias) [reply src/Opdex.Platform.Infrastructure/Data/Handlers/Addresses/Balances/SelectAddressBalancesWithFilterQueryHandler.cs:61]: `request.Cursor.TokenType == TokenProvisionalFilter.All || request.Cursor.TokenType == TokenProvisionalFilter.NonProvisional`

When filtering by all, includeLpTokens should be true as well

Comment (drmathias) [reply src/Opdex.Platform.Infrastructure/Data/Handlers/Addresses/Balances/SelectAddressBalancesWithFilterQueryHandler.cs:61]: Well, actually when filtering by all `isLpt` is unused. Guess you're right it should just be `request.Cursor.TokenType == TokenProvisionalFilter.NonProvisional`

Comment (drmathias) [reply src/Opdex.Platform.Infrastructure/Data/Handlers/Addresses/Balances/SelectAddressBalancesWithFilterQueryHandler.cs]: Ahh you mean `== TokenProvisionalFilter.Provisional`, that's right I've updated it

**Comment by drmathias** [comment]
@mrtpain Do you need to do any front end changes before rolling this out?

**Pull request merged**